apps/functions-subscription/src/domain/activation.ts

```diff
@@ -0,0 +1,79 @@
+import * as TE from 'fp-ts/TaskEither';
+import * as t from 'io-ts';
+import { IsoDateFromString } from '@pagopa/ts-commons/lib/dates';
+import { NonNegativeNumber } from '@pagopa/ts-commons/lib/numbers';
+import { NonEmptyString } from '@pagopa/ts-commons/lib/strings';
+import { TrialId, TrialIdCodec, UserIdCodec } from './subscription';
+
+// a unique brand for id of document with type job
+interface ActivationJobIdBrand {
+  // use `unique symbol` here to ensure uniqueness across modules / packages
+  readonly ActivationJobId: unique symbol;
+}
+export const ActivationJobIdCodec = t.brand(
+  NonEmptyString,
+  (str): str is t.Branded<NonEmptyString, ActivationJobIdBrand> =>
+    str.length > 0,
+  'ActivationJobId',
+);
+export type ActivationJobId = t.TypeOf<typeof ActivationJobIdCodec>;
+
+export const ActivationJobItemCodec = t.strict({
+  id: ActivationJobIdCodec,
+  trialId: TrialIdCodec,
+  _etag: t.string,
+  createdAt: IsoDateFromString,
+  usersToActivate: NonNegativeNumber,
+  usersActivated: NonNegativeNumber,
+  type: t.literal('job'),
+});
+
+export type ActivationJobItem = t.TypeOf<typeof ActivationJobItemCodec>;
+
+// a unique brand for id of document with type request
+interface ActivationRequestIdBrand {
+  // use `unique symbol` here to ensure uniqueness across modules / packages
+  readonly ActivationRequestId: unique symbol;
+}
+export const ActivationRequestIdCodec = t.brand(
+  NonEmptyString,
+  (str): str is t.Branded<NonEmptyString, ActivationRequestIdBrand> =>
+    str.length > 0,
+  'ActivationRequestId',
+);
+export type ActivationRequestId = t.TypeOf<typeof ActivationRequestIdCodec>;
+
+export const ActivationRequestItemCodec = t.strict({
+  id: ActivationRequestIdCodec,
+  trialId: TrialIdCodec,
+  _etag: t.string,
+  userId: UserIdCodec,
+  createdAt: IsoDateFromString,
+  type: t.literal('request'),
+  activated: t.boolean,
+});
+export type ActivationRequestItem = t.TypeOf<typeof ActivationRequestItemCodec>;
+
+export type ActivationResult = 'success' | 'fail';
```

Comment:
**question**: I'm not sure if it's doable, but it might be more useful to know which elements failed in case of an error. Do you think it could make sense as information?

Reply:
the underlying operation return an array of status codes for every item (the status code is an HTTP status code) to tell the result of the operation.
I have to better look at the documentation if the order of the status codes reflects the order of the operation passed as input. If so, I think we could easily return a structure like
```typescript
[{
  status: number,
  id: string
}]
```
where the `status` is the HTTP status code related to the operation and the `id` is the identifier of the resource